assets/aml-benchmark/components/src/benchmark_result_aggregator/main.py

```diff
@@ -0,0 +1,239 @@
+]:
+    """Get the details of the pipeline."""
+    runs = get_all_runs_in_current_experiment()
+    current_run = Run.get_context()
+    pipeline_params: Dict[str, Dict[str, Any]] = {}
+    loggable_params: Dict[str, Any] = {}
+    model_name = None
+    model_version = None
+    model_registry = None
+    for run in runs:
+        run_info = run.info
+        run_id = run_info.run_id
+        run_v1 = Run(experiment=current_run.experiment, run_id=run_id)
+        step_name = get_step_name(run)
+        run_details = cast(Dict[str, Any], run_v1.get_details())
+        logger.info(f"Run details for {run_id}: {run_details}")
+        run_definition = run_details.get('runDefinition', None)
+        run_v1_properties = run_details.get('properties', None)
+        logger.info(f"Run definition for {run_id}: {run_definition}")
+
+        step_params: Dict[str, Any] = {
+            'inputs': {},
+            'param': {},
+            'run_id': run_id,
+            'start_time': run_info.start_time,
+            'end_time': run_info.end_time,
+            'status': run_info.status,
+            'maxRunDurationSeconds': run_details.get('maxRunDurationSeconds', None),
+            'is_reused': False,
+        }
+
+        if run_v1_properties.get('azureml.isreused', None) == 'true':
+            step_params['is_reused'] = True
+            step_params['reused_run_id'] = \
+                run_v1_properties.get('azureml.reusedrunid', None)
+            step_params['reused_pipeline_runid'] = \
+                run_v1_properties.get('azureml.reusedpipelinerunid', None)
+
+        if run_definition is not None:
+            # Log input assets
+            input_assets = run_definition.get('inputAssets', None)
+            if input_assets is not None:
+                for key, value in input_assets.items():
+                    step_params['inputs'][key] = {
+                        'assetId': value['asset']['assetId'],
+                        'type': value['asset']['type'],
+                        'mechanism': value['mechanism'],
+                    }
+                    if step_params['inputs'][key]['type'] == 'MLFlowModel':
+                        if model_name is None:
+                            # We found a mlflow model in the pipeline
+                            model_uri: str = step_params['inputs'][key]['assetId']
+                            model_name, model_version, model_registry = \
+                                get_mlflow_model_name_version(model_uri)
+                        else:
+                            # If an mlflow model is found multiple times,
+                            # we don't know which one was benchmarked.
+                            # Hence, handle ambiguous case by not logging model.
+                            model_name = None
+                            model_version = None
+                            model_registry = None
```

Comment:
We may want to break different sections into different methods.

Reply:
I have refactored the function to multiple smaller functions to get various configs separately. Thanks!